contrib/epee/include/time_helper.h

```diff
@@ -28,132 +28,60 @@
 
 #pragma once
 
-//#include <atltime.h>
-//#include <sqlext.h>
-#include <boost/date_time/posix_time/posix_time.hpp>
-#include <boost/date_time/local_time/local_time.hpp>
-#include "pragma_comp_defs.h"
+#include <chrono>
+#include <cstdio>
+#include <ctime>
+#include <string>
 
 namespace epee
 {
 namespace misc_utils
 {
-
-#ifdef __ATLTIME_H__
-
-	inline
-	bool get_time_t_from_ole_date(DATE src, time_t& res)
+	inline bool get_gmt_time(time_t t, struct tm &tm)
 	{
-		SYSTEMTIME st = {0};
-		if(TRUE != ::VariantTimeToSystemTime(src, &st))
-			return false;
-		ATL::CTime ss(st);
-		res = ss.GetTime();
-		return true;
-	}
+#ifdef _WIN32
+		return gmtime_s(&tm, &t);
+#else
+		return gmtime_r(&t, &tm);
 #endif
-	inline 
-	std::string get_time_str(const time_t& time_)
-	{
-
-
-		char tmpbuf[200] = {0};
-		tm* pt = NULL;
-PRAGMA_WARNING_PUSH
-PRAGMA_WARNING_DISABLE_VS(4996)
-		pt = localtime(&time_);
-PRAGMA_WARNING_POP
-
-		if(pt)
-			strftime( tmpbuf, 199, "%d.%m.%Y %H:%M:%S", pt );
-		else
-		{
-			std::stringstream strs;
-			strs << "[wrong_time: " << std::hex << time_ << "]";
-			return strs.str();
-		}
-		return tmpbuf;
 	}
 
-	inline 
-		std::string get_time_str_v2(const time_t& time_)
-	{
-
-		char tmpbuf[200] = {0};
-		tm* pt = NULL;
-PRAGMA_WARNING_PUSH
-PRAGMA_WARNING_DISABLE_VS(4996)
-		pt = localtime(&time_);
-PRAGMA_WARNING_POP
-
-		if(pt)
-			strftime( tmpbuf, 199, "%Y_%m_%d %H_%M_%S", pt );
-		else
-		{
-			std::stringstream strs;
-			strs << "[wrong_time: " << std::hex << time_ << "]";
-			return strs.str();
-		}
-		return tmpbuf;
-	}
-
-  inline 
-    std::string get_time_str_v3(const boost::posix_time::ptime& time_)
-  {
-    return boost::posix_time::to_simple_string(time_);
-  }
-
-  
-
 	inline std::string get_internet_time_str(const time_t& time_)
 	{
 		char tmpbuf[200] = {0};
-		tm* pt = NULL;
-PRAGMA_WARNING_PUSH
-PRAGMA_WARNING_DISABLE_VS(4996)
-		pt = gmtime(&time_);
-PRAGMA_WARNING_POP
-		strftime( tmpbuf, 199, "%a, %d %b %Y %H:%M:%S GMT", pt );
+		struct tm pt;
+		get_gmt_time(time_, pt);
+		strftime(tmpbuf, 199, "%a, %d %b %Y %H:%M:%S GMT", &pt);
 		return tmpbuf;
 	}
 
 	inline std::string get_time_interval_string(const time_t& time_)
 	{
-		std::string res;
 		time_t tail = time_;
-PRAGMA_WARNING_PUSH
-PRAGMA_WARNING_DISABLE_VS(4244)
-		int days = tail/(60*60*24);
+		const int days = (int) (tail/(60*60*24));
```

Comment:
In C++ you'd use `static_cast<int>(tail/(60*60*24);` for better clarity and to avoid warnings.

Reply:
Good point, see the most recent commit. Thanks for reviewing!

Reply:
N.P. The latest commit looks good in this regard. 

I'm only 25% done for now. I shall continue tomorrow.